conf/varnish.vcl-example

```diff
@@ -0,0 +1,47 @@
+# A really basic example to use the Varnish api key checking.
+# Copy this to /etc/varnish/default.vcl to use it (or edit
+# /etc/default/varnishto point at this)
+import std;
+include "/etc/varnish/varnish-apikey.vcl";
+
+backend local_django {
+  .host = "127.0.0.1";
+  .port = "8000";
+}
+
+#
+# Subroutine that defines the api, key and token.
+#
+sub recognize_apiname_apikey_token {
+  # Identify api
+
+  # As mapit is both an API and a website, we only want to invoke the api key
+  # checking on the actual api endpoints. We purposely exclude the .html
+  # versions of the endpoints too, because they're basically documentation,
+  # not an actual use of the API.
+  if (req.url ~ "^/postcode/.*[^\.html$]" ||
```

Comment:
These regexes aren't correct – this says match anything as long as the last character matched isn't one of the characters in the string `\.html$` (so it would match `/postcode/hey.html` with the match `/postcode/hey`). You want `^\/postcode\/.*(?<!\.html)$` (match anything as long as the end isn't `.html`).